lib/rest/RequestHandler.js

```diff
@@ -57,10 +58,37 @@ class RequestHandler {
     */
     request(method, url, auth, body, file, _route, short) {
         const route = _route || this.routefy(url, method);
+        return this._request(method, url, auth, body, file, route, short).then(
+            (res) => res,
+            (rej) => {
+                if(!(rej instanceof DiscordHTTPError || rej instanceof DiscordRESTError)) {
+                    ++this.ratelimits[route].remaining;
+                }
+                throw rej;
+            }
+        );
+    }
```

Comment:
I'm wondering if it's better to just chain this back into the original `request` method or if it's fine